frame/name-service/src/lib.rs

```diff
@@ -0,0 +1,351 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2017-2020 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! A simple name service that can be used to give accounts friendly names.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use sp_std::prelude::*;
+use sp_runtime::traits::{Saturating};
+use frame_support::{decl_module, decl_error, decl_event, decl_storage, ensure, RuntimeDebug};
+use frame_support::dispatch::DispatchResult;
+use frame_support::traits::{
+	Currency, ReservableCurrency, Get, EnsureOrigin, OnUnbalanced,
+	WithdrawReason, ExistenceRequirement::KeepAlive, Imbalance,
+};
+use frame_system::ensure_signed;
+use codec::{Encode, Decode};
+
+mod mock;
+mod tests;
+mod benchmarking;
+
+type BalanceOf<T> = <<T as Trait>::Currency as Currency<<T as frame_system::Trait>::AccountId>>::Balance;
+type NegativeImbalanceOf<T> = <<T as Trait>::Currency as Currency<<T as frame_system::Trait>::AccountId>>::NegativeImbalance;
+
+pub trait WeightInfo {}
+
+/// The module's config trait.
+pub trait Trait: frame_system::Trait {
+
+	/// The currency trait.
+	type Currency: ReservableCurrency<Self::AccountId>;
+
+	/// The overarching event type.
+	type Event: From<Event<Self>> + Into<<Self as frame_system::Trait>::Event>;
+
+	/// Origin that can have high level control over the name-service pallet.
+	type ManagerOrigin: EnsureOrigin<Self::Origin>;
+
+	/// Origin that can set permanent ownership of a name to an account.
+	type PermanenceOrigin: EnsureOrigin<Self::Origin>;
+
+	/// Time available between subsequent bids for a name.
+	type BiddingPeriod: Get<Self::BlockNumber>;
+
+	/// Time available after bidding has completed for the winner to claim their name.
+	type ClaimPeriod: Get<Self::BlockNumber>;
+
+	/// One ownership period, which can be multiplied through exponential deposit.
+	type OwnershipPeriod: Get<Self::BlockNumber>;
+
+	/// Handler for the unbalanced decrease when funds are burned.
+	type BurnDestination: OnUnbalanced<NegativeImbalanceOf<Self>>;
+
+	/// Minimum length for a publicly available name.
+	type MinLength: Get<usize>;
```

Comment:
what's the reason to use `usize`? Shouldn't we want to avoid them because it is different between wasm and native?

Reply:
I dont think usize here is an issue, and it is just less code to write when checking against the len of a name. In either case I removed this now since I think it is better to use a hash instead of the name directly.